import-export-cli/impl/deleteThrottlingPolicy.go

```diff
@@ -0,0 +1,139 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package impl
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+)
+
+// DeleteThrottlingPolicy
+// @param accessToken : Access Token for the resource
+// @param policyName : Name of the Throttling Policy to delete
+// @param policyType : Type of the Throttling Policy to delete
+// @param environment : Environment where Throttling should be deleted
+// @return response Response in the form of *resty.Response
+func DeleteThrottlingPolicy(accessToken, policyName, policyType, environment string) (*resty.Response, error) {
+	endpoint := utils.GetAdminEndpointOfEnv(environment, utils.MainConfigFilePath)
+	endpoint = utils.AppendSlashToString(endpoint)
+
+	var throttlingPolicyType string
+
+	resource := "throttling/policies/search"
+	searchEndpoint := endpoint + resource
+
+	// queryParamString := `query=name:` + policyName
+
+	switch policyType {
+	case CmdPolicyTypeSubscription:
+		throttlingPolicyType = QueryPolicyTypeSubscription
+	case CmdPolicyTypeApplication:
+		throttlingPolicyType = QueryPolicyTypeApplication
+	case CmdPolicyTypeAdvanced:
+		throttlingPolicyType = QueryPolicyTypeAdvanced
+	case CmdPolicyTypeCustom:
+		throttlingPolicyType = QueryCmdPolicyTypeCustom
+	}
+
+	queryParamString := `query=name:` + policyName + ` type:` + throttlingPolicyType
+
+	url := searchEndpoint
+
+	policyId, err := getThrottlingPolicyId(accessToken, environment, url, queryParamString, policyName)
+
+	if policyId == "" && err == nil {
+		errMsg := "Requested Policy with name=" + policyName + " and type=" + policyType + " not found."
+		utils.HandleErrorAndExit("Deletion Failed! ", errors.New(errMsg))
+	}
+
+	if err != nil {
+		utils.HandleErrorAndExit("Error while getting Throttling Policy Id for deletion ", err)
+	}
+
+	resource = "throttling/policies/"
+
+	switch policyType {
+	case CmdPolicyTypeSubscription:
+		resource += utils.ThrottlingPolicyTypeSub
+	case CmdPolicyTypeApplication:
+		resource += utils.ThrottlingPolicyTypeApp
+	case CmdPolicyTypeAdvanced:
+		resource += utils.ThrottlingPolicyTypeAdv
+	case CmdPolicyTypeCustom:
+		resource += utils.ThrottlingPolicyTypeCus
+	}
+
+	resource = utils.AppendSlashToString(resource)
+
+	resource += policyId
```

Comment:
Why can't we add the policyId in the previous line itself?
resource = utils.AppendSlashToString(resource) + policyId

Reply:
Fixed by [7f99f9c](https://github.com/wso2/product-apim-tooling/pull/927/commits/7f99f9c6a4a514e1c644f81ab6efd235ac4f5f20)